imports/plugins/core/core/server/method-hooks.js

```diff
@@ -99,27 +98,30 @@ MethodHooks._initializeHook = function (mapping, methodName, hookFunction) {
     // Call the main method body
     // check(args, Match.Any);
     try {
-      methodResult = MethodHooks._originalMethodHandlers[methodName].apply(this, args);
+      methodResult = Promise.await(MethodHooks._originalMethodHandlers[methodName].apply(this, args));
     } catch (error) {
       methodError = error;
     }
 
     // Call after hooks, providing the result and the original arguments
-    const afterHooks = MethodHooks._afterHooks[methodName];
-    _.each(afterHooks, (afterHook, hooksProcessed) => {
-      const hookResult = afterHook.call(this, {
+    const afterHooks = MethodHooks._afterHooks[methodName] || [];
+    hooksProcessed = 0;
+    for (const afterHook of afterHooks) {
+      const hookResult = Promise.await(afterHook.call(this, {
         result: methodResult,
         error: methodError,
         arguments: args,
         hooksProcessed
-      });
+      }));
       // If the after hook did not return a value and the methodResult is not undefined, warn and fix
-      if (_.isUndefined(hookResult) && !_.isUndefined(methodResult)) {
+      if (hookResult === undefined && methodResult !== undefined) {
         Meteor._debug("Expected the after hook to return a value.");
```

Comment:
We don't usually use `Meteor._debug`. Change this to `Logger.warn` while you're in here.